WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsTimeRangeCard.swift

```diff
@@ -0,0 +1,61 @@
+import SwiftUI
+
+struct AnalyticsTimeRangeCard: View {
+
+    let timeRangeTitle: String
+    let currentRangeDescription: String
+    let previousRangeDescription: String
+
+    var body: some View {
+        VStack(alignment: .leading, spacing: Layout.verticalSpacing) {
+            HStack {
+                ZStack(alignment: .center) {
+                    Circle()
+                        .fill(Color(UIColor.systemGray6))
+                        .frame(width: Layout.calendarCircleWidth)
+                    Image(uiImage: .calendar)
+                }
+                VStack(alignment: .leading, spacing: .zero) {
+                    Text(timeRangeTitle)
+                        .foregroundColor(Color(.text))
+                        .subheadlineStyle()
+                    Text(currentRangeDescription)
+                        .bold()
+                }
+                .padding(.leading)
+                .frame(maxWidth: .infinity, alignment: .leading)
+            }
+            .padding(.leading)
+
+            Divider()
+
+            BoldableTextView(String.localizedStringWithFormat(Localization.previousRangeComparisonContent, previousRangeDescription))
```

Comment:
nit: You can add the `String.localizedStringWithFormat...` as a function in the localization menu to keep the view layout code clear!

Reply:
Makes perfect sense. I've adjusted here: https://github.com/woocommerce/woocommerce-ios/pull/8159/commits/6f8a7a13234aff0eb83ee7edad3147ba6d98de6f

Let me know if you think it makes sense this way.